spec/services/measurements_creator_spec.rb

```diff
@@ -0,0 +1,60 @@
+require "spec_helper"
+
+describe MeasurementsCreator do
+  let(:measurements_attributes) { [{:longitude=>25.4356212, :latitude=>56.4523456, :time=>"2016-05-11T17:09:02", :milliseconds=>925, :measured_value=>59.15683475380729, :value=>59.15683475380729}] }
+
+  it "creates a mesurement" do
+    stream = create_stream!
+    MeasurementsCreator.call(stream, measurements_attributes)
+
+    expect(Measurement.count).to eq(1)
+  end
+
+  it "creates a mesurement with utc time for fixed sessions" do
+    stream = create_stream!("FixedSession")
+    MeasurementsCreator.call(stream, measurements_attributes)
+
+    expect(Measurement.first.utc_time).to be_within(1.second).of Time.now
+  end
+
+  it "creates a mesurement without utc time for mobile sessions" do
+    stream = create_stream!("MobileSession")
+    MeasurementsCreator.call(stream, measurements_attributes)
+
+    expect(Measurement.first.utc_time).to be_nil
+  end
+
+  private
+
+  def create_stream!(session_type  = "MobileSession")
```

Comment:
Have you considered doing something like this (https://github.com/HabitatMap/AirCasting/blob/staging/spec/services/csv/export_sessions_to_csv_spec.rb#L71)?

That way you have the same api of active_record (you pass a hash) and all the defaults. That way you are explicit with the important stuff `session_type` and hide all of the noise

Reply:
I like the idea of having the same api. I could do something like that:
```
session = create_session!(type: "FixedSession")
stream = create_stream!(session: session)
```
but I am not sure about passing all the defaults, they make noise that's irrelevant to this test case.